### ISSUE TITLE
Selecting the best model while training

### ISSUE THREAD

**Issue by vanhoanglepsa** [logpai/deep-loglizer#1]
Hi. Thanks for your project. It's really helpful.

I'm running DeepLog (lstm) model on the HDFS dataset. I found that the longer I train, the worse I get. For example, at epoch 5, the f1 is 0.917 with k = 4 and training loss = 0.71088. However, from epoch 9 to 18, when the training loss is around 0.53, the best f1 is only 0.778. I thought that the smaller loss will lead to better results. Is that right?

I wonder is there any way to choose the best model based on the training values instead of looking at the results on the testing set?
https://github.com/logpai/deep-loglizer/blob/ca50c2860bb0602d087a1109e5c27bfd5fc70477/deeploglizer/models/base_model.py#L310

Thank you.

**Comment by zbchern** [comment]
Hi, sorry for the late reply.

A smaller training loss does not necessarily lead to a better result. We believe this is caused by overfitting, since HDFS is a relatively simple dataset.

To alleviate this issue, you can try decreasing the complexity of the model, e.g., using a smaller setting of hidden_size or num_layers. Thanks.

**Comment by trungkien210493** [comment]
@vanhoanglepsa could you share to me the parameters you used to reach f1 = 0.917. I'm tuning the input parameters but just reach the f1 around 0.7, less than very much compare to the paper.

**Comment by vanhoanglepsa** [comment]
@trungkien210493 I remember I only set the batch size to 512 and keep the default setting for other parameters. you can set num_layers to 1 and hidden_size to 64 to decrease the model complexity.

**Comment by trungkien210493** [comment]
@vanhoanglepsa Thank you

**Comment by trungkien210493** [comment]
![image](https://user-images.githubusercontent.com/43126932/148503725-f1ad67f6-9a10-4bab-a9c5-0b24c6a5e695.png)
@vanhoanglepsa are there any mystics ? I tried to use the batch_size with "lstm_demo.py" but reach the f1= 0.6897 only. I also try to run the command line in benchmarks folder but the same result

**Comment by zbchern** [comment]
> ![image](https://user-images.githubusercontent.com/43126932/148503725-f1ad67f6-9a10-4bab-a9c5-0b24c6a5e695.png) @vanhoanglepsa are there any mystics ? I tried to use the batch_size with "lstm_demo.py" but reach the f1= 0.6897 only. I also try to run the command line in benchmarks folder but the same result

Hi. Using the default setting provided in the benchmarks folder, I can easily obtain an f1 score of 0.929 (at the third epoch). Are you using the full dataset or just the demo one?

**Comment by trungkien210493** [comment]
> > ![image](https://user-images.githubusercontent.com/43126932/148503725-f1ad67f6-9a10-4bab-a9c5-0b24c6a5e695.png) @vanhoanglepsa are there any mystics ? I tried to use the batch_size with "lstm_demo.py" but reach the f1= 0.6897 only. I also try to run the command line in benchmarks folder but the same result
> 
> Hi. Using the default setting provided in the benchmarks folder, I can easily obtain an f1 score of 0.929 (at the third epoch). Are you using the full dataset or just the demo one?

Hi. I am using the code in "demo" folder, which use the dataset "/data/processed/HDFS_100k/hdfs_0.0_tar". @zbchern , as you mention, does the f1 score of 0.929 is the f1 score for full log ???

**Comment by zbchern** [comment]
> > > ![image](https://user-images.githubusercontent.com/43126932/148503725-f1ad67f6-9a10-4bab-a9c5-0b24c6a5e695.png) @vanhoanglepsa are there any mystics ? I tried to use the batch_size with "lstm_demo.py" but reach the f1= 0.6897 only. I also try to run the command line in benchmarks folder but the same result
> > 
> > 
> > Hi. Using the default setting provided in the benchmarks folder, I can easily obtain an f1 score of 0.929 (at the third epoch). Are you using the full dataset or just the demo one?
> 
> Hi. I am using the code in "demo" folder, which use the dataset "/data/processed/HDFS_100k/hdfs_0.0_tar". @zbchern , as you mention, does the f1 score of 0.929 is the f1 score for full log ???

@

**Comment by zbchern** [comment]
> > > ![image](https://user-images.githubusercontent.com/43126932/148503725-f1ad67f6-9a10-4bab-a9c5-0b24c6a5e695.png) @vanhoanglepsa are there any mystics ? I tried to use the batch_size with "lstm_demo.py" but reach the f1= 0.6897 only. I also try to run the command line in benchmarks folder but the same result
> > 
> > 
> > Hi. Using the default setting provided in the benchmarks folder, I can easily obtain an f1 score of 0.929 (at the third epoch). Are you using the full dataset or just the demo one?
> 
> Hi. I am using the code in "demo" folder, which use the dataset "/data/processed/HDFS_100k/hdfs_0.0_tar". @zbchern , as you mention, does the f1 score of 0.929 is the f1 score for full log ???

Hi. In this case, you are using the demo dataset. Yes, the 0.929 f1 score comes from the full log.
